core/engine/src/object/builtins/lazy_builtin.rs

```diff
@@ -0,0 +1,260 @@
+        (builtin.init)(realm);
+    }
+
+    ordinary_define_own_property(obj, key, desc, context)
+}
+
+pub(crate) fn lazy_has_property(
+    obj: &JsObject,
+    key: &PropertyKey,
+    context: &mut InternalMethodContext<'_>,
+) -> JsResult<bool> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_has_property(obj, key, context)
+}
+
+pub(crate) fn lazy_try_get(
+    obj: &JsObject,
+    key: &PropertyKey,
+    receiver: JsValue,
+    context: &mut InternalMethodContext<'_>,
+) -> JsResult<Option<JsValue>> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_try_get(obj, key, receiver, context)
+}
+
+pub(crate) fn lazy_get(
+    obj: &JsObject,
+    key: &PropertyKey,
+    receiver: JsValue,
+    context: &mut InternalMethodContext<'_>,
+) -> JsResult<JsValue> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_get(obj, key, receiver, context)
+}
+
+pub(crate) fn lazy_set(
+    obj: &JsObject,
+    key: PropertyKey,
+    value: JsValue,
+    receiver: JsValue,
+    context: &mut InternalMethodContext<'_>,
+) -> JsResult<bool> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_set(obj, key, value, receiver, context)
+}
+
+pub(crate) fn lazy_delete(
+    obj: &JsObject,
+    key: &PropertyKey,
+    context: &mut InternalMethodContext<'_>,
+) -> JsResult<bool> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_delete(obj, key, context)
+}
+
+pub(crate) fn lazy_own_property_keys(
+    obj: &JsObject,
+    context: &mut Context,
+) -> JsResult<Vec<PropertyKey>> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    if builtin.is_initialized.get() {
+        (builtin.init)(realm);
+    }
+
+    ordinary_own_property_keys(obj, context)
+}
+
+pub(crate) fn lazy_call(
+    obj: &JsObject,
+    argument_count: usize,
+    context: &mut Context,
+) -> JsResult<CallValue> {
+    let realm = context.realm();
+    let builtin = obj.downcast_ref::<Builtin>().expect("obj is not a Builtin");
+    (builtin.init)(realm);
+
+    if let BuiltinKind::Constructor(constructor) = &builtin.kind {
+        return Ok(constructor.__call__(argument_count));
+    }
+
+    non_existant_call(obj, argument_count, context)
```

Comment:
This should not be checked at runtime, but inside the `JsData` implementation. The reason is that setting an ordinary object's construct or call function to something different to `non_existant_data` will incorrectly determine that the object is callable.